cpp/src/io/utilities/file_io_utilities.cpp

```diff
@@ -166,28 +168,67 @@ void cufile_registered_file::register_handle()
 cufile_registered_file::~cufile_registered_file() { shim->handle_deregister(cf_handle); }
 
 cufile_input_impl::cufile_input_impl(std::string const& filepath)
-  : shim{cufile_shim::instance()}, cf_file(shim, filepath, O_RDONLY | O_DIRECT)
+  : shim{cufile_shim::instance()}, cf_file(shim, filepath, O_RDONLY | O_DIRECT), pool(16)
```

Comment:
I thought 8 threads is where the perf benefit plateaus. Please add a comment on the chosen number.

Reply:
Actually 8 -> 16 still gave an improvement of a couple of ms when going from creating threads in each read call to pool of threads. Before the pool, the thread creation overhead overpowered it. The difference I saw after more than 16 threads was just noise.